pkg/mount/magic.go

```diff
@@ -0,0 +1,212 @@
+	SQUASHFS = []byte{'h', 's', 'q', 's'}
+	VFAT     = []byte{0xeb, 0x58}
+	XFS      = []byte{'X', 'F', 'S', 'B'}
+
+	AAFS        = []byte{0x5a, 0x3c, 0x69, 0xf0}
+	ADFS        = []byte{0xad, 0xf5}
+	AFFS        = []byte{0xad, 0xff}
+	AFS         = []byte{0x53, 0x46, 0x41, 0x4F}
+	BDEVFS      = []byte{0x62, 0x64, 0x65, 0x76}
+	BINDERFS    = []byte{0x6c, 0x6f, 0x6f, 0x70}
+	BINFMTFS    = []byte{0x42, 0x49, 0x4e, 0x4d}
+	BPF         = []byte{0xca, 0xfe, 0x4a, 0x11}
+	BTRFS       = []byte{0x91, 0x23, 0x68, 0x3E}
+	CGROUP      = []byte{0x27, 0xe0, 0xeb}
+	CGROUP2     = []byte{0x63, 0x67, 0x72, 0x70}
+	CODA        = []byte{0x73, 0x75, 0x72, 0x45}
+	CRAMFS      = []byte{0x28, 0xcd, 0x3d, 0x45}
+	CRAMFSOther = []byte{0x45, 0x3d, 0xcd, 0x28}
+	DAXFS       = []byte{0x64, 0x64, 0x61, 0x78}
+	DEBUGFS     = []byte{0x64, 0x62, 0x67, 0x20}
+	DEVPTS      = []byte{0x1c, 0xd1}
+	ECRYPTFS    = []byte{0xf1, 0x5f}
+	EFIVARFS    = []byte{0xde, 0x5e, 0x81, 0xe4}
+	EFS         = []byte{0x41, 0x4A, 0x53}
+	// EXFAT seems to be a samsung file system.
+	//EXFAT       = []byte{0x53, 0xef}
+	F2FS      = []byte{0xF2, 0xF5, 0x20, 0x10}
+	FUSE      = []byte{0x65, 0x73, 0x55, 0x46}
+	FUTEXFS   = []byte{0xBA, 0xD1, 0xDE, 0xA}
+	HOSTFS    = []byte{0x00, 0xc0, 0xff, 0xee}
+	HPFS      = []byte{0xf9, 0x95, 0xe8, 0x49}
+	HUGETLBFS = []byte{0x95, 0x84, 0x58, 0xf6}
+	JFFS2     = []byte{0x72, 0xb6}
+	JFS       = []byte{0x31, 0x53, 0x46, 0x4a}
+	MTD       = []byte{0x11, 0x30, 0x78, 0x54}
+	NFS       = []byte{0x69, 0x69}
+	NILFS     = []byte{0x34, 0x34}
+	NSFS      = []byte{0x6e, 0x73, 0x66, 0x73}
+	// From docs, not tested.
+	NTFS       = []byte{0xeb, 0x52, 0x90, 'N', 'T', 'F', 'S', ' ', ' ', ' ', ' '}
+	OCFS2      = []byte{0x74, 0x61, 0x63, 0x6f}
+	OPENPROM   = []byte{0x9f, 0xa1}
+	OVERLAYFS  = []byte{0x79, 0x4c, 0x76, 0x30}
+	PIPEFS     = []byte{0x50, 0x49, 0x50, 0x45}
+	PROC       = []byte{0x9f, 0xa0}
+	PSTOREFS   = []byte{0x61, 0x65, 0x67, 0x6C}
+	QNX4       = []byte{0x00, 0x2f}
+	QNX6       = []byte{0x68, 0x19, 0x11, 0x22}
+	RAMFS      = []byte{0x85, 0x84, 0x58, 0xf6}
+	RDTGROUP   = []byte{0x76, 0x55, 0x82, 1}
+	ROMFS      = []byte{0x72, 0x75}
+	SECURITYFS = []byte{0x73, 0x63, 0x66, 0x73}
+	SELINUX    = []byte{0xf9, 0x7c, 0xff, 0x8c}
+	SMACK      = []byte{0x43, 0x41, 0x5d, 0x53}
+	SMB        = []byte{0x51, 0x7B}
+	SOCKFS     = []byte{0x53, 0x4F, 0x43, 0x4B}
+	SYSFS      = []byte{0x62, 0x65, 0x65, 0x72}
+	TMPFS      = []byte{0x01, 0x02, 0x19, 0x94}
+	TRACEFS    = []byte{0x74, 0x72, 0x61, 0x63}
+	UBIFS      = []byte{0x24, 0x05, 0x19, 0x05}
+	UDF        = []byte{0x15, 0x01, 0x33, 0x46}
+	USBDEVICE  = []byte{0x9f, 0xa2}
+	V9FS       = []byte{0x01, 0x02, 0x19, 0x97}
+	XENFS      = []byte{0xab, 0xba, 0x19, 0x74}
+	ZONEFS     = []byte{0x5a, 0x4f, 0x46, 0x53}
+	ZSMALLOC   = []byte{0x58, 0x29, 0x58, 0x29}
+)
+
+type magic struct {
+	magic []byte
+	off   int
+	name  string
+	flags uintptr
+}
+
+// magics maps the statfs fsword (an ambiguous type, to say the least) to strings
```

Comment:
Why can't we use the statfs syscall?

Reply:
it only works on mounted fs. Which you told me. And I had forgotten. It's pretty useless for this purpose, sadly.

Reply:
for reasons you told me last night :)